data/content_scripts/popup.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @fileOverview This script is injected in popup.html and is used to 
+ * communicate with the add-on code by making use of the Jetpack messaging API,
+ * self.port.
+ */
+
+// Receive the "extensionMode" message from the add-on.
+// Change the label in popup.html accordingly.
+self.port.on("extensionMode", function(message) {
+  if (typeof message === "string") {
```

Comment:
Separate the event handler from the `self.port.on` so you can name it and document it better.

You should also write a document to keep track of all the messages that are being sent and received by the extension. This namespace is going to be getting crowded and it is best to make sure there are not collisions.